### ISSUE TITLE
Quaternion freezes on start on M1 Mac after brew install

### ISSUE THREAD

**Issue by jakecoppinger** [quotient-im/Quaternion#759]
### Description

Quaternion freezes on start on an M1 MacbookPro after `brew install quaternion`. I'm not sure if this platform has been tested yet - just wanted to let you know in case you didn't :) Please let me know if I'm doing something silly.

### Steps to reproduce

- `brew install quaternion`
- Right click on `.app` then click open (bypass macOS security policy)
- App asks for notification permission
- No window or menubar appears
- Right click on app icon in Dock says "Application Not Responding"

### Version information

- **Quaternion version**: `0.0.9.4e`
- **Install method**:  Homebrew
- **Platform**: MacOS 11.2 (M1 CPU)

**Comment by KitsuneRal** [comment]
There doesn't seem to be anything silly per se. Can you start the application from the terminal window? It should write some things there as it runs.

**Comment by aaronraimist** [comment]
I think this is probably a Qt bug. When 0.0.9.4e is run with `env QT_LOGGING_RULES="*.debug=true"` on an M1 Mac it gets stuck in a loop like this when it says "Application Not Responding":

```
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow"))>
qt.qpa.drawing: [QNSView displayLayer] QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow")
qt.qpa.drawing: QCocoaWindow::handleExposeEvent QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClassWindow") QRegion(0,0 282x247) isExposed true
qt.qpa.backingstore: Flushing QRegion(0,0 282x247) of <QNSView: 0x7f9415e0c490; QCocoaWindow(0x7f9415e0c320, window=QWidgetWindow(0x7f9415e0bdf0, name="LoginDialogClass^C
```

When 0.0.9.4e is built with Qt 5.15.2 it works fine on an M1 Mac. 0.0.95-beta4 also works. So depending on the ETA for 0.0.95 stable we can either wait for that or build 0.0.9.4f for Mac and get that into Homebrew.

**Comment by KitsuneRal** [comment]
Thanks for the investigation. I intend to make 0.0.95 beta 5 today; apparently I should switch CI to 5.15.2 before that. If there’s nothing wrong with the beta the release is likely to follow in 2-3 weeks.

**Comment by jakecoppinger** [comment]
Thanks very much for investigating, sorry about not getting back to you on running from the terminal.

**Comment by KitsuneRal** [comment]
Does the .dmg at https://github.com/quotient-im/Quaternion/actions/runs/658939586 work fine?

**Comment by mhmdanas** [comment]
I am facing this same issue on an Intel Mac. I was going to try the DMG that @KitsuneRal posted, but it doesn't seem to be available anymore. Building from source works fine.

**Comment by aaronraimist** [comment]
@mhmdanas what version are you using when you reproduce the bug? 0.0.9.4 with any letter after it is quite old at this point. This bug should be fixed in the current release candidate 0.0.95 RC2.

**Comment by mhmdanas** [comment]
Indeed, the version that is live on Homebrew is 0.0.9.4e. I'll give the RC a try.
